src/main/java/net/minecraftforge/common/IPlantable.java

```diff
@@ -23,26 +23,22 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockReader;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+
 public interface IPlantable
 {
+    @Nullable
     default PlantType getPlantType(IBlockReader world, BlockPos pos) {
-        if (this instanceof AttachedStemBlock) return PlantType.FARMLAND;
-        if (this instanceof CropsBlock) return PlantType.FARMLAND;
-        if (this instanceof DeadBushBlock) return PlantType.DEAD_BUSH;
-        if (this instanceof FungusBlock) return PlantType.FUNGUS;
-        if (this instanceof LilyPadBlock) return PlantType.LILY_PAD;
-        if (this instanceof MushroomBlock) return PlantType.MUSHROOM;
-        if (this instanceof NetherRootsBlock) return PlantType.NYLIUM_PLANTS;
-        if (this instanceof NetherSproutsBlock) return PlantType.NYLIUM_PLANTS;
-        if (this instanceof NetherWartBlock) return PlantType.NETHER_WART;
-        if (this instanceof SeaGrassBlock) return PlantType.SEA_GRASS;
-        if (this instanceof SeaPickleBlock) return PlantType.SEA_PICKLE;
-        if (this instanceof StemBlock) return PlantType.FARMLAND;
-        if (this instanceof TallSeaGrassBlock) return PlantType.SEA_GRASS;
-        if (this instanceof WitherRoseBlock) return PlantType.WITHER_ROSE;
-
-        if (this instanceof BushBlock) return PlantType.PLAINS;
-        throw new IllegalStateException("The class implementing IPlantable has no PlantType!");
+        for (Map.Entry<String, PlantType> plantType : PlantType.VALUES.entrySet()) {
+            if (plantType.getValue().getAllowedPlants().size() == 0) {
```

Comment:
This looks wrong to me, it will cause the first plant type with no allowed blocks to match for all plants. Means all I have to do is add a plant type with no allowed values and all other plant types fail.

I would also suggest instead of this messy looking conditonal, add something like `PlantType::contains(BlockState)`

Reply:
Yeah, the default implementation should only return plant types if they are in a set of allowed blocks. Otherwise, if the set of allowed blocks is empty, then it should be up to implementers of this method to return that plant type instead.